File: src/main/java/org/trp/shincolle/entity/base/ShipCombatDecisionResolver.java
package org.trp.shincolle.entity.base;

final class ShipCombatDecisionResolver {
    private ShipCombatDecisionResolver() {
    }

    static boolean shouldClearAfterStuck(ShipMovementRecoveryState recovery) {
        return ShipRecoveryDecisionResolver.shouldClearAfterStuck(
                recovery.stuckTicks(), ShipAiNumbers.PASSIVE_COMBAT_STUCK_TICK_LIMIT);
    }

    static boolean shouldClearAfterMoveFailures(int failCount) {
        return ShipRecoveryDecisionResolver.shouldClearAfterMoveFailures(
                failCount, ShipAiNumbers.PASSIVE_COMBAT_MOVE_FAIL_LIMIT);
    }
}

File: src/main/java/org/trp/shincolle/entity/base/ShipCombatDecisionResolver.kt
package org.trp.shincolle.entity.base

internal object ShipCombatDecisionResolver {
    fun shouldClearAfterStuck(recovery: ShipMovementRecoveryState): Boolean {
        return ShipRecoveryDecisionResolver.shouldClearAfterStuck(
            recovery.stuckTicks(), ShipAiNumbers.PASSIVE_COMBAT_STUCK_TICK_LIMIT
        )
    }

    fun shouldClearAfterMoveFailures(failCount: Int): Boolean {
        return ShipRecoveryDecisionResolver.shouldClearAfterMoveFailures(
            failCount, ShipAiNumbers.PASSIVE_COMBAT_MOVE_FAIL_LIMIT
        )
    }
}
